src/scenicplus/cistromes.py

```diff
@@ -314,6 +314,93 @@ def TF_cistrome_correlation(scplus_obj: SCENICPLUS,
+            gene expression from
+            `scplus_obj.uns['Pseudobulk'][variable]['Cistromes_AUC'][cistromes_key][signature_key2]`.
+        out_key : str, optional
+            Output key. Correlations will be stored at `scplus_obj.uns['cistrome_correlation'][variable][out_key]`.
+        subset: List, optional
+            Subset of cells to be used to calculate correlations. Default: None (All)
+        """
+        if use_pseudobulk:
+            gene_auc_agg = scplus_obj.uns['Pseudobulk'][variable][auc_key][signature_key1]
+            region_auc_agg = scplus_obj.uns['Pseudobulk'][variable][auc_key][signature_key2]
+        else:
+            gene_auc_agg = scplus_obj.uns[auc_key][signature_key1].copy().T
+            region_auc_agg = scplus_obj.uns[auc_key][signature_key2].copy().T
+
+        if subset is not None:
+            cell_data = pd.DataFrame([x.rsplit('_', 1)[0] for x in gene_auc_agg.columns],
+                                     index=gene_auc_agg.columns).iloc[:, 0]
+            subset_cells = cell_data[cell_data.isin(subset)].index.tolist()
+            gene_auc_agg = gene_auc_agg.loc[:, subset_cells]
+            region_auc_agg = region_auc_agg.loc[:, subset_cells]
+
+        corr_df = pd.DataFrame(columns=['Signature1', 'Signature2', 'Rho', 'P-value'])
+
+        # cistrome naming includes number of genes/regions, so need to create matching names
+        # x.rsplit('_', 1) splits at first _ from the right
+        gene_auc_agg['cistrome_id_short'] = gene_auc_agg.index.map(lambda x: x.rsplit('_', 1)[0])
+        gene_auc_agg['cistrome_id'] = gene_auc_agg.index
+        gene_auc_agg = gene_auc_agg.set_index('cistrome_id_short')
+
+        region_auc_agg['cistrome_id_short'] = region_auc_agg.index.map(lambda x: x.rsplit('_', 1)[0])
+        region_auc_agg['cistrome_id'] = region_auc_agg.index
+        region_auc_agg = region_auc_agg.set_index('cistrome_id_short')
+
+        for tf in gene_auc_agg.index:
+            # I think all TFs should match, but just in case
+            if tf in region_auc_agg.index:
+                gene_auc_agg_tf = gene_auc_agg.loc[tf, gene_auc_agg.columns != 'cistrome_id']
+                region_auc_agg_tf = region_auc_agg.loc[tf, gene_auc_agg.columns != 'cistrome_id']
+
+                # record orginal cistrome name for results
+                signature1_id = gene_auc_agg.loc[tf, 'cistrome_id']
+                signature2_id = region_auc_agg.loc[tf, 'cistrome_id']
+                # Exception in case TF is only expressed in 1 cell
+                # TFs expressed in few cells could be filtered too
+                try:
+                    corr_1, _1 = pearsonr(gene_auc_agg_tf, region_auc_agg_tf)
+                    x = {'Signature1': signature1_id,
+                         'Signature2': signature2_id,
+                         'Rho': corr_1,
+                         'P-value': _1}
+                    corr_df = pd.concat([corr_df,
```

Comment:
I get a lot of warnings from `corr_df.append`: ` FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.`

I switched to `pd.concat` here as suggested. I think it would probably be more efficient to create the output as a list of data frames and then concatenate them all in one step, though.